cpp/open3d/core/nns/FixedRadiusSearch.cu

```diff
@@ -512,28 +512,209 @@ void WriteNeighborsIndicesAndDistances(
     }
 }
 
-/// Kernel for MaxKnnThreshold
-template <class T>
-__global__ void MaxKnnThresholdKernel(
-        const int64_t* const __restrict__ prev_indices,
-        const T* const __restrict__ prev_distances,
+/// Kernel for WriteNeighborsHybrid
+template <class T, int METRIC, bool RETURN_DISTANCES>
+__global__ void WriteNeighborsHybridKernel(
         int64_t* __restrict__ indices,
         T* __restrict__ distances,
-        const int64_t* const __restrict__ neighbors_counts,
-        const int64_t* const __restrict__ neighbors_row_splits,
-        int64_t num_queries,
-        int max_knn) {
+        const int64_t* const __restrict__ point_index_table,
+        const int64_t* const __restrict__ hash_table_cell_splits,
+        size_t hash_table_size,
+        const T* const __restrict__ query_points,
+        size_t num_queries,
+        const T* const __restrict__ points,
+        size_t num_points,
+        const T inv_voxel_size,
+        const T radius,
+        const T threshold,
+        const int max_knn) {
     int query_idx = blockDim.x * blockIdx.x + threadIdx.x;
     if (query_idx >= num_queries) return;
 
-    int64_t indices_offset = neighbors_row_splits[query_idx];
-    int64_t num_neighbors = neighbors_counts[query_idx];
+    int count = 0;  // counts the number of neighbors for this query point
+
+    size_t indices_offset = max_knn * query_idx;
 
-    for (int j = 0; j < max_knn; j++) {
-        indices[max_knn * query_idx + j] =
-                j >= num_neighbors ? -1 : prev_indices[indices_offset + j];
-        distances[max_knn * query_idx + j] =
-                j >= num_neighbors ? -1 : prev_distances[indices_offset + j];
+    Vec3<T> query_pos(query_points[query_idx * 3 + 0],
+                      query_points[query_idx * 3 + 1],
+                      query_points[query_idx * 3 + 2]);
+    Vec3<int> voxel_index = ComputeVoxelIndex(query_pos, inv_voxel_size);
+    int hash = SpatialHash(voxel_index) % hash_table_size;
+
+    int bins_to_visit[8] = {hash, -1, -1, -1, -1, -1, -1, -1};
+
+    for (int dz = -1; dz <= 1; dz += 2)
```

Comment:
let's add the full `{}` for the for loops

Reply:
Done.